artichoke-backend/src/extn/core/symbol/mod.rs

```diff
@@ -1,115 +1,14 @@
 use std::ffi::c_void;
 
 use crate::convert::{Immediate, UnboxedValueGuard};
-use crate::extn::core::array::Array;
 use crate::extn::prelude::*;
-use crate::intern::Symbol as SymbolId;
 
 pub mod ffi;
 pub mod mruby;
 pub mod trampoline;
 
-#[derive(Debug, Clone, Copy, Hash, PartialEq, Eq, PartialOrd, Ord)]
-pub struct Symbol(SymbolId);
-
-impl Symbol {
-    pub fn all_symbols(interp: &mut Artichoke) -> Result<Array, Exception> {
-        let all_symbols = interp.all_symbols()?.map(Symbol::from).collect::<Vec<_>>();
-        let array = all_symbols
-            .into_iter()
-            .map(|symbol| Symbol::alloc_value(symbol, interp))
-            .collect::<Result<Vec<Value>, _>>()?;
-        Ok(Array::from(array))
-    }
-
-    #[inline]
-    #[must_use]
-    pub fn id(self) -> SymbolId {
-        self.0
-    }
-
-    #[must_use]
-    pub fn is_empty(self, interp: &mut Artichoke) -> bool {
-        if let Ok(Some(bytes)) = interp.lookup_symbol(self.into()) {
-            bytes.is_empty()
-        } else {
-            true
-        }
-    }
-
-    #[must_use]
-    pub fn len(self, interp: &mut Artichoke) -> usize {
-        if let Ok(Some(bytes)) = interp.lookup_symbol(self.into()) {
-            bytes.len()
-        } else {
-            0_usize
-        }
-    }
-
-    #[must_use]
-    pub fn bytes<'a>(&self, interp: &'a mut Artichoke) -> &'a [u8] {
```

Comment:
this method in `spinoso-symbol` takes `self` by value and eliminates the explicit lifetime parameters.